constructor-manager/src/constructor_manager/utils/request.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from functools import lru_cache
+from typing import List
+
+import requests
+from constructor_manager import __version__
+from constructor_manager.defaults import DEFAULT_TIMEOUT
+
+logger = logging.getLogger(__name__)
+
+
+@lru_cache
+def _user_agent() -> str:
+    """Return a user agent string for use in http requests.
+
+    Returns
+    -------
+    str
+        User agent string.
+    """
+    return f"constructor-manager-{__version__}"
+
+
+def get_request(url: str) -> requests.Response:
+    """Return a requests session with a user agent.
+
+    Parameters
+    ----------
+    url : str
+        URL to request.
+
+    Returns
+    -------
+    requests.Response
+        Response object.
+    """
+    session = requests.Session()
+    session.headers.update({"user-agent": _user_agent()})
+    return session.get(url, timeout=DEFAULT_TIMEOUT)
+
+
+@lru_cache
+def plugin_versions(
+    url: str,
+) -> List[str]:
+    """Return information on package plugins from endpoint in json.
```

Comment:
I guess we will need to document this API generally?

Reply:
Absolutely, will add the docs